flash/core/model.py

```diff
@@ -108,20 +108,22 @@ def step(self, batch: Any, batch_idx: int) -> Any:
         output["y"] = y
         return output
 
-    def forward(self, x: Any) -> Any:
-        return self.model(x)
+    def forward(self, *args: Any, **kwargs: Any) -> Any:
+        return self.model(*args, **kwargs)
 
-    def training_step(self, batch: Any, batch_idx: int) -> Any:
-        output = self.step(batch, batch_idx)
+    # Mypy requires the steps to have this signature since the base class has it and a more explicit
+    # phrasing of the arguments would violate the Liskov substitution principle
```

Comment:
I prefer if we keep the explicit arguments, and adding `noqa`  if necessary. We want users subclassing this to not have to google the actual arguments they will receive etc.

If this was user code, It would be okay but it's not the case.

Reply:
Fine with this as well. But we will have many of these then :)

Reply:
I know... but that's the case too in Lightning. Not sure if there is a better solution